KsApi/models/ExportDataEnvelope.swift

```diff
@@ -2,21 +2,46 @@
 
 public struct ExportDataEnvelope {
   public let expiresAt: String?
-  public let state: State
+  public var state: State {
+    if let stateString, let state = State(rawValue: stateString) {
+      return state
+    }
+    return .unknown
```

Comment:
Yea this is good, can be made a little cleaner:
```
guard let stateRawValue, let state = State(rawValue: stateRawValue) else {
return .unknown
    }

    return state
```

I was confused at first because it looked like a pure `getter`, which wouldn't allow the `Lenses` to set the `state` property as they were doing before, but including the initializer for both `state` sets `state` via `stateString`, which works for the lenses.

I also now understand  that we aren't switching on the enum, simply setting its' initial state.

Reply:
Updated to use `guard let` instead :) Glad we're on the same page, now!